scripts/internal_links_monitor.py

```diff
@@ -0,0 +1,243 @@
+                                    'SAI_PORT_STAT_IF_OUT_ERRORS']
+        # Using fixed values for 201911
+        self.threshold = 0
+        self.mitigationActionEnabled = True
+
+    @property
+    def namespaces(self):
+        namespaces = []
+        all_namespaces = multi_asic.get_all_namespaces()
+        namespaces = all_namespaces['front_ns'] + all_namespaces['back_ns']
+        return namespaces
+        
+    def config_db(self,namespace):
+        '''Returns the config db object for the namespace'''
+        db = SonicV2Connector(namespace=namespace)
+        db.connect('CONFIG_DB')
+        return db
+    
+    def appl_db(self, namespace):
+        '''Returns the application db object for the namespace'''
+        db = SonicV2Connector(namespace=namespace)
+        db.connect('APPL_DB')
+        return db
+
+    def get_port_status(self, namespace, port_name):
+        ''' Returns operational status of give port 
+            When port is admin shut in CONFIG_DB, it might take some time for port to be oper down in APPL_DB.
+            As such, status of ports which are admin shut in config DB is reported as down
+        '''
+        port_cfg = self.configdb[namespace].get_all('CONFIG_DB', 'PORT|{}'.format(port_name))
+        port_info = self.appdb[namespace].get_all('APPL_DB', 'PORT_TABLE:{}'.format(port_name))
+        if port_cfg:
+            if port_cfg['admin_status'] == 'up':
+                if port_info:
+                    return port_info['oper_status']
+        return 'down'
+
+    def filter_for_backend_ports(self, namespace, port_map):
+        ''' Returns only internal (backend) ports which are operationally up for monitoring '''
+        filtered_port_map = {}
+        for k,v in port_map.items():
+            if k.startswith('Ethernet-BP'):                
+                if self.get_port_status(namespace, k) == 'up':
+                    filtered_port_map[k] = v
+        return filtered_port_map
+
+    def get_lag_name_for_port(self, namespace, port_name):
+        ''' Returns name of the portchannel in which the given port is a member '''
+        lag_member = self.appdb[namespace].keys('APPL_DB', 'LAG_MEMBER_TABLE:*:{}'.format(port_name))
+        if len(lag_member):
+            table = lag_member[0].find(":") + 1
+            port = lag_member[0].find(":", table)
+            return lag_member[0][table:port]
+        return ''
+
+    def get_min_links_for_lag(self, namespace, lag_name):
+        ''' Returns min_links configuration of the given portchannel '''
+        lag_info = self.configdb[namespace].get_all('CONFIG_DB', 'PORTCHANNEL|{}'.format(lag_name))        
+        return int(lag_info['min_links'])        
+
+    def get_active_lag_member_count(self, namespace, lag_name):
+        ''' Returns number of member ports that are operationally up in the given portchannel '''
+        lag_members = self.appdb[namespace].keys('APPL_DB', 'LAG_MEMBER_TABLE:{}:*'.format(lag_name))
```

Comment:
can we use config db instead of appl db ?

Reply:
since we are looking for active (operationally up) lag members, does it matter if we check in appl db?